app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user (not logged in)
+
+    can :destroy, Post, author_id: user.id # Allow users to delete their own posts
+    can :destroy, Comment, author_id: user.id # Allow users to delete their own comments
+
+    return unless user.role == 'admin' # Check if the user has an admin role
+
+    can :destroy, Post # Allow admins to delete any post
+    can :destroy, Comment # Allow admins to delete any comment
+  end
```

Comment:
- [ ] You have done an excellent job, but here you are giving the admin and user the right only to destroy, but no right to read, like and create a comment, also admin should be able to manage all activities. so you can look at this code snippet and revamp your method
```
 if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
```

Reply:
Added as instructed